### PR TITLE
feat: aimbot

### PR DESCRIPTION
**Pull request by iconFehu**
update aimbot

### CODE REVIEW

**Comment by Seanghost117** [comment]
I'm sorry but what is it you are even doing? You are altering a lot of stuff with the source in a sloppy and messy way. You are messing with version control on various repos, you are not following the code format and structure. You left a bunch of commented code and are using your own repo (I assume) for classes instead of the official one. You are making and jamming in classes and functions into various spots. You are hardcoding in CN text, and messing with the ci...

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,20 +43,20 @@ set_property(TARGET YimMenu PROPERTY CXX_STANDARD 23) # 23 Because std::format i
...
+target_include_directories(YimMenu PRIVATE
     "${SRC_DIR}"
     "${json_SOURCE_DIR}/single_include"
     "${gtav_classes_SOURCE_DIR}"
     "${imgui_SOURCE_DIR}"
 )
 
 target_precompile_headers(YimMenu PRIVATE "${SRC_DIR}/common.hpp")
-target_link_libraries(YimMenu PRIVATE pugixml minhook AsyncLogger dbghelp imgui cpr Zydis)
+target_link_libraries(YimMenu PRIVATE pugixml minhook AsyncLogger dbghelp imgui cpr Zydis ${OpenCV_LIBS})
```
Comment (Seanghost117) [inline CMakeLists.txt]: why add opencv and in this format?

File: scripts/cpr.cmake [hunk 2]
```diff
@@ -5,7 +5,7 @@ set(CURL_DISABLE_TESTS OFF)
 FetchContent_Declare(
     cpr
     GIT_REPOSITORY https://github.com/libcpr/cpr.git
-    GIT_TAG        1986262ba4e0cb052161e9e7919aef5ef08217f0
+    GIT_TAG        1.10.1
```
Comment (Seanghost117) [inline scripts/cpr.cmake]: Why mess with the current git tags, they are the way they are for a reason

File: scripts/json.cmake [hunk 3]
```diff
@@ -5,7 +5,7 @@ set(JSON_MultipleHeaders OFF)
 FetchContent_Declare(
     json
     GIT_REPOSITORY https://github.com/ArthurSonzogni/nlohmann_json_cmake_fetchcontent.git
-    GIT_TAG        67e6070f9d9a44b4dec79ebe6b591f39d2285593
+    GIT_TAG        master
```
Comment (Seanghost117) [inline scripts/json.cmake]: again this is messing with the version control and can cause issues later down the road

File: scripts/minhook.cmake [hunk 4]
```diff
@@ -2,8 +2,8 @@ include(FetchContent)
 
 FetchContent_Declare(
     minhook
-    GIT_REPOSITORY https://github.com/YimMenu/minhook.git
-    GIT_TAG        902ab63e0a771547961e132ccc0700d62e2a1423
+    GIT_REPOSITORY https://github.com/TsudaKageyu/minhook.git
```
Comment (Seanghost117) [inline scripts/minhook.cmake]: Please use the yim branch for custom scripts not someone else's or your own branch

File: scripts/opencv.cmake [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#include(FetchContent)
+#
+#message("opencv")
+#
+#FetchContent_Declare(
```
Comment (Seanghost117) [inline scripts/opencv.cmake]: Why include all this commented code?

File: scripts/pugixml.cmake [hunk 6]
```diff
@@ -3,7 +3,7 @@ include(FetchContent)
 FetchContent_Declare(
     pugixml
     GIT_REPOSITORY https://github.com/zeux/pugixml.git
-    GIT_TAG        a0e064336317c9347a91224112af9933598714e9
+    GIT_TAG        master
```
Comment (Seanghost117) [inline scripts/pugixml.cmake]: Again for here and zydis, version control

File: src/backend/looped/weapons/aimbot.cpp [hunk 7]
```diff
@@ -3,25 +3,69 @@
...
+		int type{};
+		Vector3 world_position;
+		Vector2 screen_dim{}, movement{};
 
-		virtual void on_tick() override
+	public:
+		void on_tick() override
 		{
-			float local_fov_change = g.weapons.aimbot.fov;
+			if (kf.is_first_tick)
```
Comment (Seanghost117) [inline src/backend/looped/weapons/aimbot.cpp]: on_enable is a thing, no reason to do this method

File: src/backend/looped/weapons/aimbot.cpp [hunk 8]
```diff
@@ -97,10 +140,38 @@ namespace big
...
+
+									//									LOG(INFO) << kf.prediction.at<float>(0) << " " << kf.prediction.at<float>(1);
+
+									//									kf.prediction.at<float>()
+									//									auto& io = ImGui::GetIO();
+									//									ImVec2 center(io.DisplaySize.x / 2.f, io.DisplaySize.y / 2.f);
+									//auto predicted_pos = ImVec2(center.x + prediction.at<float>(
+
+									//									cv::Mat prediction          = kf.kalman_filter.predict();
+									//									cv::Mat corrected           = kf.kalman_filter.correct(kf.measurement);
```
Comment (Seanghost117) [inline src/backend/looped/weapons/aimbot.cpp]: why leave so much commented code here?

File: src/pointers.cpp [hunk 9]
```diff
@@ -803,6 +809,22 @@ namespace big
 			memory::byte_patch::make(handle.add(6).as<uint16_t*>(), 0x9090)->apply();
 		}
 
+		std::map<std::wstring, std::wstring> baseStringList = {
+			{L"isCollimation",L"GTA5.exe+20DE250"},
```
Comment (Seanghost117) [inline src/pointers.cpp]: but why?

File: src/services/notifications/notification_service.cpp [hunk 10]
```diff
@@ -6,6 +6,16 @@ namespace big
...
+		// Dumping stacktrace:
+		// [0]     KERNELBASE.dll RaiseException
+		// [1]     D:\a\_work\1\s\src\vctools\crt\vcruntime\src\eh\throw.cpp L: 75 _CxxThrowException
+		// [2]     C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\format L: 79 std::_Throw_format_error
+		// [3]     C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\format L: 942 std::_Fmt_codec<char,0>::_Find_encoded
+		// [4]     C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\format L: 1293 std::_Parse_format_string<char,std::_Format_handler<char> &>
+		// [5]     C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\format L: 3422 std::vformat_to<std::back_insert_iterator<std::basic_string<char,std::char_traits<char>,std::allocator<char> > > >
+		// [6]     C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\format L: 3493 std::vformat<0>
+		// [7]     C:\Users\fehu\source\repos\YimMenu\src\services\notifications\notification_service.cpp L: 9 big::notification_service::notification_service
```
Comment (Seanghost117) [inline src/services/notifications/notification_service.cpp]: why are you leaving a comment with an output error / stack trace, also be aware this has your pc user name in it...

File: src/views/esp/view_esp.cpp [hunk 11]
```diff
@@ -167,11 +171,179 @@ namespace big
...
+
+				if (distance < g.esp.tracer_render_distance[1] && distance > g.esp.tracer_render_distance[0] && g.esp.tracer)
+				{
+					draw_list->AddLine({(float)*g_pointers->m_resolution_x * g.esp.tracer_draw_position[0],
+					                       (float)*g_pointers->m_resolution_y * g.esp.tracer_draw_position[1]},
+					    {esp_x, esp_y},
+					    esp_color);
+				}
+
+				//				LOG(INFO) << "ped: " << ped << " ped->m_health: " << ped->m_health << " ped->m_maxhealth: " << ped->m_maxhealth << " health_perc: " << health_perc << " (ped): " << (ped)
```
Comment (Seanghost117) [inline src/views/esp/view_esp.cpp]: Again same thing with the commented code

File: src/views/esp/view_esp.cpp [hunk 12]
```diff
@@ -167,11 +171,179 @@ namespace big
...
+
+
+				if (distance < g.esp.box_render_distance[1] && distance > g.esp.box_render_distance[0] && g.esp.box)
+				{
+					draw_list->AddRect({esp_x - (62.5f * multplr), esp_y - (175.f * multplr)}, {esp_x - (62.5f * multplr) + (125.f * multplr), esp_y - (175.f * multplr) + (350.f * multplr)}, esp_color);
+				}
+				if (g.esp.name)
+				{
+					name_str = "NPC";
+					name_str += std::format(" | 速度{}", ped->get_speed());
```
Comment (Seanghost117) [inline src/views/esp/view_esp.cpp]: Please make sure the text is in english

File: src/pointers.cpp [hunk 13]
```diff
@@ -746,6 +746,12 @@ namespace big
 			m_migrate_object = ptr.as<functions::migrate_object>();
 		});
 
+		// camGameplayDirectorPointer
+		main_batch.add("camGameplayDirector", "48 8B 05 ? ? ? ? 38 98 ? ? ? ? 8A C3", [this](memory::handle ptr)
+		{
+			camGameplayDirector = ptr.add(3).rip().as<uintptr_t>();
```
Comment (TheGreenBandit) [inline src/pointers.cpp:752]: please use proper format m_cam_gameplay_director

File: src/views/esp/view_esp.cpp [hunk 14]
```diff
@@ -167,11 +171,179 @@ namespace big
 		}
 	}
 
+
+	void esp::draw_ped(CPed* ped, ImDrawList* const draw_list)
```
Comment (maybegreat48) [inline src/views/esp/view_esp.cpp]: This function isn't being used and should be removed

File: src/pointers.cpp [hunk 15]
```diff
@@ -803,6 +809,22 @@ namespace big
...
 
+		std::map<std::wstring, std::wstring> baseStringList = {
+			{L"isCollimation",L"GTA5.exe+20DE250"},
+			{L"collimationPed",L"GTA5.exe+20DE240"},
+			{L"START_SHAPE_TEST_CAPSULE_FUN",L"GTA5.exe+D5C442"},
+			{L"GET_SHAPE_TEST_RESULT_FUN",L"GTA5.exe+D42E28"},
+		};
+
+		auto BaseAddress = (uintptr_t)GetModuleHandle("GTA5.exe");
+		isCollimation = BaseAddress+0x20DE250;
```
Comment (maybegreat48) [inline src/pointers.cpp]: This "collimation" thing isn't being used anywhere

**Review by maybegreat48 — COMMENTED** [review]
What does your PR actually do to the aimbot? Why are you calling into a massive AI library every frame?

**Pull request closed without merging**